test/core/test_lwt.ml

```diff
@@ -1801,7 +1801,56 @@ let async_tests = suite "async" [
       !saw = Some Exception)
   end;
 ]
-let suites = suites @ [async_tests]
+let dont_wait_tests = suite "dont_wait" [
+  test "fulfilled" begin fun () ->
+    let f_ran = ref false in
+    Lwt.dont_wait (fun () -> f_ran := true; Lwt.return ()) (fun _ -> ());
+    later (fun () -> !f_ran = true)
+  end;
+
+  test ~sequential:true "f raises" begin fun () ->
+    let saw = ref None in
+    Lwt.dont_wait
+      (fun () -> raise Exception)
+      (fun exn -> saw := Some exn);
+    later (fun () ->
+      !saw = Some Exception)
+  end;
+
+  test ~sequential:true "rejected" begin fun () ->
+    let saw = ref None in
+    Lwt.dont_wait
+      (fun () -> Lwt.fail Exception)
+      (fun exn -> saw := Some exn);
+    later (fun () ->
+      !saw = Some Exception)
+  end;
+
+  test "pending, fulfilled" begin fun () ->
+    let resolved = ref false in
+    let p, r = Lwt.wait () in
+    Lwt.dont_wait
+      (fun () ->
+        Lwt.bind p (fun () ->
+          resolved := true;
+          Lwt.return ()))
+      (fun _ -> ());
+    Lwt.wakeup r ();
+    later (fun () -> !resolved = true)
+  end;
+
+  test ~sequential:true "pending, rejected" begin fun () ->
+    let saw = ref None in
+    let p, r = Lwt.wait () in
+    Lwt.dont_wait
+      (fun () -> p)
+      (fun exn -> saw := Some exn) ;
+    Lwt.wakeup_exn r Exception;
+    later (fun () ->
+      !saw = Some Exception)
+  end;
+]
+let suites = suites @ [async_tests ; dont_wait_tests]
```

Comment:
I recommend not accumulating two suites on one line, i.e. leave old line 1804 alone. The idea is to do the concatenation "locally" to each test suite, so that the concatenations don't get lost, and we don't accidentally stop running some tests when code is edited.